src/persist/src/indexed/encoding.rs

```diff
@@ -111,20 +107,39 @@ pub struct StreamRegistration {
     pub val_codec_name: String,
 }
 
-/// The metadata necessary to reconstruct an Unsealed.
+/// The metadata necessary to reconstruct an Arrangement.
 ///
 /// Invariants:
-/// - The batch SeqNo ranges are sorted and non-overlapping.
+/// - The unsealed_batch SeqNo ranges are sorted and non-overlapping.
+/// - The trace_batch Descriptions are sorted, non-overlapping, and contiguous.
+/// - The since frontier is either 0 or < the trace's sealed frontier.
+/// - Every batch's since frontier is <= the overall trace's since frontier.
+/// - The compaction level of trace_batches is weakly decreasing when iterating
+///   from oldest to most recent time intervals.
+/// - Every trace_batch's upper is <= the overall trace's seal frontier.
+/// - TODO: key uniqueness invariants?
```

Comment:
i think this todo is a bit state -- we already enforce that all blob keys are globally unique across all arrangements in `BlobMeta`

Reply:
:facepalm: Totally blanked on this, I'll fix it in the next PR.